libff/algebra/curves/bls12_381/bls12_381_g1.hpp

```diff
@@ -0,0 +1,126 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef BLS12_381_G1_HPP_
+#define BLS12_381_G1_HPP_
+#include <libff/algebra/curves/bls12_381/bls12_381_init.hpp>
+#include <libff/algebra/curves/curve_utils.hpp>
+#include <vector>
+
+namespace libff
+{
+
+class bls12_381_G1;
+std::ostream &operator<<(std::ostream &, const bls12_381_G1 &);
+std::istream &operator>>(std::istream &, bls12_381_G1 &);
+
+class bls12_381_G1
+{
+public:
+#ifdef PROFILE_OP_COUNTS
+    static long long add_cnt;
+    static long long dbl_cnt;
+#endif
+    static std::vector<std::size_t> wnaf_window_table;
+    static std::vector<std::size_t> fixed_base_exp_window_table;
+    static bls12_381_G1 G1_zero;
+    static bls12_381_G1 G1_one;
+    static bls12_381_Fq coeff_a; // VV
+    static bls12_381_Fq coeff_b; // VV
+
+    // Cofactor
+    static const mp_size_t h_bitcount = 126;
+    static const mp_size_t h_limbs =
+        (h_bitcount + GMP_NUMB_BITS - 1) / GMP_NUMB_BITS;
+    static bigint<h_limbs> h;
+
+    typedef bls12_381_Fq base_field;
+    typedef bls12_381_Fr scalar_field;
+
+    bls12_381_Fq X, Y, Z;
+
+    // using Jacobian coordinates
+    bls12_381_G1();
+    bls12_381_G1(
+        const bls12_381_Fq &X, const bls12_381_Fq &Y, const bls12_381_Fq &Z)
+        : X(X), Y(Y), Z(Z){};
+
+    void print() const;
+    void print_coordinates() const;
+
+    void to_affine_coordinates();
+    void to_special();
+    bool is_special() const;
+
+    bool is_zero() const;
+
+    bool operator==(const bls12_381_G1 &other) const;
+    bool operator!=(const bls12_381_G1 &other) const;
+
+    bls12_381_G1 operator+(const bls12_381_G1 &other) const;
+    bls12_381_G1 operator-() const;
+    bls12_381_G1 operator-(const bls12_381_G1 &other) const;
+
+    bls12_381_G1 add(const bls12_381_G1 &other) const;
+    bls12_381_G1 mixed_add(const bls12_381_G1 &other) const;
+    bls12_381_G1 dbl() const;
+    bls12_381_G1 mul_by_cofactor() const;
+
+    // from bls12_377 (VV)
```

Comment:
I think all these old comments can safely be removed.

Reply:
Done.